### ISSUE TITLE
Using Blob or SafeResourceUrl images

### ISSUE THREAD

**Issue by default23** [lukasz-galka/ngx-gallery#21]
Hello, i'm stuck with displaying images with SafeResourceURL or Blob images, i don't have a direct link to the image like  'https://example.com/image.png', because i need to get an image from the API using several headers in the request

Please leave an example or describe how can i show gallery with that images

**Comment by lukasz-galka** [comment]
Hi @default23, could you give me more info or maybe some example? Could you give me maybe example with simple html img?

**Comment by default23** [comment]
Hi, i'm receive images with method like this

```typescript
getFile(fileName, id): Observable<Blob> {

        return Observable.create((obs: Observer<Blob>) => {

            const token: string     = localStorage.getItem(JWT);
            const xhr: XMLHttpRequest = new XMLHttpRequest();
            xhr.responseType        = 'blob';

            xhr.onreadystatechange = function () {
                if (xhr.readyState === 4) {
                    xhr.status === 200 ? obs.next(xhr.response) : obs.error(null);
                    obs.complete();
                }
            };

            xhr.open('GET', `${PROXY_MANAGER}/files/${id}/${fileName}`, true);
            xhr.setRequestHeader('Authorization', 'Bearer ' + token);
            xhr.send(null);
        });
    }
```

And then, i'll create SafeResourceUrl for that blob, because Angular fail with security error, with this method

```typescript
 private _createImageUrl(image: Blob): SafeResourceUrl {

        return this._sanitizer.bypassSecurityTrustResourceUrl(
            URL.createObjectURL(image)
        );
}
```

And finally i'm getting the all images of SafeResourceUrl type, it have a link inside like

blob://localhost.....

**Comment by lukasz-galka** [comment]
Hi @default23, in version 1.1.1 I have updated `NgxGalleryImage` and now is suporting `SafeResourceUrl `
https://github.com/lukasz-galka/ngx-gallery#ngxgalleryimage

I have also added example in demo https://lukasz-galka.github.io/ngx-gallery-demo/#images-as-saferesourceurl

Please let me know if it helps.

**Comment by default23** [comment]
Ok @lukasz-galka , thanks a lot! 👍  I'll leave a feedback later

**Comment by default23** [comment]
@lukasz-galka This warning appears again

```
WARNING: sanitizing unsafe style value url(SafeValue must use [property]=binding: blob:http://localhost:4200/f8ef3afa-1204-4e91-b29f-00a003db92c2 (see http://g.co/ng/security#xss)) (see http://g.co/ng/security#xss).
```

Seems to this warning occurs from this line of your code
```html
[style.background-image]="'url(' + image + ')'"
```

**Comment by default23** [comment]
@lukasz-galka 

This warning appears again

```
WARNING: sanitizing unsafe style value url(SafeValue must use [property]=binding: blob:http://localhost:4200/6afc3ae1-ab7a-481e-b4f7-16481b0dc86c (see http://g.co/ng/security#xss)) (see http://g.co/ng/security#xss).
```

For the testing purposes i load files via this business logic

```typescript
loadImageFiles() {

        const id = this.driver.id;
        this.driver.driverFiles
            .forEach(file => this._driverService.getDriverFile(id, file.name)
                .subscribe((res: Blob) => 

                     this.galleryImages.push({
                        small : this._fileHelper.createImageUrl(res),
                        medium: this._fileHelper.createImageUrl(res),
                        big   : this._fileHelper.createImageUrl(res)
                    });
                }));
    }

// fileHelper.ts

createImageUrl(image: Blob): SafeResourceUrl {
        return this._sanitizer.bypassSecurityTrustResourceUrl(URL.createObjectURL(image));
}
```

I think this images are not render because you need to use the `bypassSecurityTrustStyle ` method (DomSanitizer) before you use an image in the HTML

`ngx-gallery-image.component.html`
```html
 [style.background-image]="'url(' + image + ')'"
```

[Issue like that on stackoverflow](https://stackoverflow.com/questions/38593515/angular2-warning-sanitizing-unsafe-style-value-urlsafevalue-must-use-proper)

**Comment by lukasz-galka** [comment]
Yes, but there is a problem because `bypassSecurityTrustStyle` requires `string` and you are using `SafeResourceUrl`. I will read about it.

**Comment by lukasz-galka** [comment]
I know that it is probably to late, but I fixed this in version `1.1.7`

**Comment by default23** [comment]
@lukasz-galka when i'll back to my project i will check this! Thank you
